ui/src/app/base/sagas/websockets.js

```diff
@@ -4,12 +4,33 @@ import { all, call, put, take, race } from "redux-saga/effects";
 import getCookie from "./utils";
 import WebSocketClient from "../../../websocket-client";
 
+/**
+ * Dynamically build a websocket url from window.location
+ *
+ */
+const buildWsUrl = () => {
+  let protocol;
```

Comment:
If you wished: `const protocol = window.location.protocol === "https:" ? "wss:" : "ws:"`.

Reply:
thanks